go/test/endtoend/multipooler/conn_cache_test.go

```diff
@@ -240,12 +240,22 @@ func TestMultiPoolerClient(t *testing.T) {
 					// Make an RPC to both services
 					_, err := client.ConsensusStatus(ctx, pooler, &consensusdatapb.StatusRequest{})
 					if err != nil && ctx.Err() == nil {
-						t.Errorf("worker %d: consensus Status RPC failed: %v", workerID, err)
+						// With grpc.NewClient() lazy connections, DeadlineExceeded can occur
+						// when the context expires right before/during connection establishment.
+						// This is expected behavior near test completion, not a real failure.
+						if !strings.Contains(err.Error(), "DeadlineExceeded") && !strings.Contains(err.Error(), "context deadline exceeded") {
```

Comment:
Any chance it's possible to check if the error or an unwrapped error is equal to `context.DeadlineExceeded` instead of comparing the string contents?